### PR TITLE
feat: use react-admin 4

### PR DESCRIPTION
**Pull request by alanpoulain**
| Q             | A
| ------------- | ---
| Branch?       | next
| Tickets       | N/A
| License       | MIT
| Doc PR        | N/A

TODO:
- [x] Try with the demo
- [x] Add an UPGRADE.md guide
- [x] Ask react-admin core team to review

### CODE REVIEW

**Comment by alanpoulain** [comment]
@fzaninotto @djhi @WiXSL could I ask one of you to have a quick look at this PR? It would be great to have your input on it!

**Comment by fzaninotto** [comment]
Hi Alan,

Great work!

We're focused on releasing an RC. When that's addressed, we'll take a look at your PR.

File: src/layout/Error.tsx [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+
+interface InternalErrorProps
+  extends Omit<HtmlHTMLAttributes<HTMLDivElement>, 'title'>,
+    Partial<Pick<FallbackProps, 'resetErrorBoundary'>>,
+    ErrorProps {
+  className?: string;
+}
+
+// eslint-disable-next-line tree-shaking/no-side-effects-in-initialization
+const Root = styled('div', {
```
Comment (fzaninotto) [inline src/layout/Error.tsx:41]: If you don't intend to make this component overiddable , don't bother with `styled`, and use `sx` diurectly in the JSX code.

**Comment by alanpoulain** [comment]
Thank you very much for the review.

**Pull request merged**